File: log_analysis/modules/storages/storage.py
import psycopg2 as pg

class ConnectionData:
    def __init__(self):
        self.host = None
        self.port = None
        self.database = None
        self.user = None
        self.password = None

class Storage:
    def __init__(self, connection_data):
        self.connection = pg.connect(host=connection_data.host, port=connection_data.port, dbname=connection_data.database, user=connection_data.user, password=connection_data.password)

    def do(self, query, should_return=False):
        transaction = self.connection.cursor()
        transaction.execute(query)

        self.connection.commit()

        if should_return:
            return transaction.fetchall()

File: log_analysis_web_backend/log_analysis/storage/src/storage.cpp
#include <sstream>

#include <storage/storage.h>

using namespace storage;

Storage& Storage::Get() {
    static Storage storage;
    return storage;
}

void Storage::Connect(const ConnectionData& connection_data) {
    std::ostringstream connection;
    connection << "dbname = " << connection_data.database
            << " user = " << connection_data.username
            << " password = " << connection_data.password
            << " host = " << connection_data.host
            << " port = " << connection_data.port;

    connection_data_ = connection.str();
}

Transaction Storage::CreateTransaction() {
    return { connection_data_ };
}
